docs/webactions.md

```diff
@@ -0,0 +1,133 @@
+2. `code`: a valid HTTP status code (default is 200 OK).
+3. `body`: a string which is either plain text or a base64 encoded string (for binary data).
+
+The controller will pass along the action-specified headers, if any, to the HTTP client when terminating the request/response. Similarly the controller will respond with the given status code when present. Lastly, the body is passed along as the body of the response. Unless a `content-type header` is declared in the action result’s `headers`, the body is passed along as is if it’s a string (or results in an error otherwise). When the `content-type` is defined, the controller will determine if the response is binary data or plain text and decode the string using a base64 decoder as needed. Should the body fail to decoded correctly, an error is returned to the caller.
+
+## HTTP Context
+
+A web action, when invoked, receives all the HTTP request information available as additional parameters to the action input argument. They are:
+
+1. `__ow_meta_verb`: the HTTP method of the request.
+2. `__ow_meta_headers`: the request headers.
+3. `__ow_meta_path`: the unmatched path of the request (matching stops after consuming the action extension).
+
+The request may not override any of the named `__ow_` parameters above; doing so will result in a failed request with status equal to 400 Bad Request.
+
+## Additional features
+
+Web actions bring some additional features that include:
+
+1. `Content extensions`: the request must specify its desired content type as one of `.json`, `.html`, `.text` or `.http`. This is done by adding an extension to the action name in the URI, so that an action `/guest/demo/hello` is referenced as `/guest/demo/hello.http` for example to receive an HTTP response back.
+2. `Projecting fields from the result`: the path that follows the action name is used to project out one or more levels of the response. For example, 
+`/guest/demo/hello.html/body`. This allows an action which returns a dictionary `{body: "..." }` to project the `body` property and directly return its string value instead. The projected path follows an absolute path model (as in XPath).
+3. `Query and body parameters as input`: the action receives query parameters as well as parameters in the request body. The precedence order for merging parameters is: package parameters, action parameters, query parameter, body parameters with each of these overriding any previous values in case of overlap . As an example `/guest/demo/hello.http?name=Jane` will pass the argument `{name: "Jane"}` to the action.
+4. `Form data`: in addition to the standard `application/json`, web actions may receive URL encoded from data `application/x-www-form-urlencoded data` as input.
+5. `Activation via multiple HTTP verbs`: a web action may be invoked via one of four HTTP methods: `GET`, `POST`, `PUT` or `DELETE`.
+
```

Comment:
you mean in addition to this example?
this is not a tutorial on how to use these new features but rather documenting what the new features are. the examples should be small and self contained. there's no magic here - they're just actions is the point. i don't think showing every possible combination really helps.
i tend to believe it's the opposite - makes the exposition confusing.

learn by trying, these are just actions.